apps/src/lab2/projects/utils.ts

```diff
@@ -75,3 +77,19 @@ export function getFileByName(
   }
   return null;
 }
+
+/**
+ * Given a map of {fileId: ProjectFile}, return the first file with the given name.
+ * @param project - The folders and files for a given project.
+ * @returns The first file that is both active and visible, or the first file.
+ */
+export function getActiveFileForProject(project: MultiFileSource) {
+  const files = Object.values(project.files);
+  const isStartMode = getAppOptionsEditBlocks() === START_SOURCES;
+  const activeFiles = files.filter(
+    f => f.active && (f.visible !== false || isStartMode)
+  );
+
+  // If the only active file is hidden, make the first file active.
+  return activeFiles?.[0] || files[0];
```

Comment:
will this throw if `files` is empty? It's possible to have no files in a project if the user deletes all files.

Reply:
If we render a useful "placeholder" screen if no (active?) files exist, it might be okay for `getActiveFileForProject` to return null (?).